### ISSUE TITLE
dm-tool failing assertions noisily .... (GLib-CRITICAL) ... 'is_valid_heap_iter (iter)' fails

### ISSUE THREAD

**Issue by bernd-wechner** [CanonicalLtd/lightdm#39]
I'll let this trace speak for itself:

```
$ dm-tool --version
lightdm 1.26.0
$ dm-tool list-seats                                       
Seat0
  CanSwitch=true
  HasGuestAccount=false
  Session2
    UserName='bernd2'
  Session0
    UserName='bernd'

(process:10106): GLib-CRITICAL **: 15:40:02.408: g_variant_iter_free: assertion 'is_valid_heap_iter (iter)' failed

(process:10106): GLib-CRITICAL **: 15:40:02.409: g_variant_iter_free: assertion 'is_valid_heap_iter (iter)' failed
```

and for the record inxi reports:

```
System:    Kernel: 4.15.0-32-generic x86_64 bits: 64 Console: tty 14 Distro: Linux Mint 19 Tara
Machine:   Device: desktop System: ASUS product: All Series serial: N/A
           Mobo: ASUSTeK model: MAXIMUS VII HERO v: Rev 1.xx serial: N/A
           UEFI [Legacy]: American Megatrends v: 2401 date: 02/24/2015
CPU:       Quad core Intel Core i7-4790 (-MT-MCP-) cache: 8192 KB
           clock speeds: max: 4000 MHz 1: 1276 MHz 2: 1052 MHz 3: 1025 MHz 4: 1026 MHz
           5: 1043 MHz 6: 1031 MHz 7: 1053 MHz 8: 1047 MHz
Graphics:  Card: Intel Xeon E3-1200 v3/4th Gen Core Processor Integrated Graphics Controller
           Display Server: X.org 1.19.6 drivers: modesetting (unloaded: fbdev,vesa)
           tty size: 100x82 Advanced Data: N/A out of X
Audio:     Card-1 Intel Xeon E3-1200 v3/4th Gen Core Processor HD Audio Controller
           driver: snd_hda_intel
           Card-2 Intel 9 Series Family HD Audio Controller driver: snd_hda_intel
           Card-3 Logitech G930 driver: USB Audio
           Sound: Advanced Linux Sound Architecture v: k4.15.0-32-generic
Network:   Card-1: Intel Ethernet Connection (2) I218-V driver: e1000e
           IF: eno1 state: up speed: 100 Mbps duplex: full mac: 08:62:66:4a:31:5b
           Card-2: Belkin F7D2102 802.11n N300 Micro Wireless Adapter v3000 [Realtek RTL8192CU]
           driver: rtl8192cu
           IF: wlx08863bf3e29b state: N/A mac: N/A
Drives:    HDD Total Size: 3688.8GB (76.3% used)
           ID-1: /dev/sda model: KINGSTON_SV300S3 size: 240.1GB
           ID-2: /dev/sdb model: SAMSUNG_SSD_830 size: 128.0GB
           ID-3: /dev/sdc model: WDC_WD10EACS size: 1000.2GB
           ID-4: /dev/sdd model: Maxtor_6V320F0 size: 320.1GB
           ID-5: /dev/sde model: WDC_WD20EARX size: 2000.4GB
Partition: ID-1: / size: 188G used: 17G (10%) fs: ext4 dev: /dev/dm-0
           ID-2: /home size: 294G used: 126G (46%) fs: ext4 dev: /dev/sdd1
           ID-3: swap-1 size: 34.23GB used: 0.00GB (0%) fs: swap dev: /dev/dm-1
RAID:      No RAID devices: /proc/mdstat, md_mod kernel module present
Sensors:   System Temperatures: cpu: 29.8C mobo: 27.8C
           Fan Speeds (in rpm): cpu: 0
Info:      Processes: 475 Uptime: 11 days Memory: 17219.3/32044.4MB Init: systemd runlevel: 5
           Client: Shell (bash) inxi: 2.3.56
```

if that helps diagnose.

**Comment by robert-ancell** [comment]
Fixed in 6b98b30199e7a12c48a27c3a31171b02bf0d4de5, thanks!

**Comment by bernd-wechner** [comment]
Thank you! Now to push that build out to Ubuntu repos ;-)
